src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -78,12 +84,23 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
     }
   }
 
-  private def extractFee(tx: ErgoTransaction): Long = {
-    val propositionBytes = settings.chainSettings.monetary.feePropositionBytes
+  private def extractFee(tx: ErgoTransaction): Long =
     ErgoState.boxChanges(Seq(tx))._2
       .filter(_.ergoTree == settings.chainSettings.monetary.feeProposition)
       .map(_.value)
       .sum
+
+  private def randSliceIndexes(qty: Int, max: Int): Seq[Int] = {
+    require(qty <= max)
+    val idx = Random.nextInt(max)
+    @scala.annotation.tailrec
+    def loop(acc: Seq[Int], idx: Int, leftQty: Int): Seq[Int] =
+      if (leftQty > 0) {
+        if (idx <= max) loop(acc :+ idx, idx + 1, leftQty - 1) else loop(acc, 0, leftQty)
+      } else {
+        acc
+      }
+    loop(Array.empty[Int], idx, qty)
```

Comment:
@oskin1 `val res = idx to (idx + qty)`?

Reply:
@aslesarenko, And what if we want to select `10` transactions having `100` transactions in the mempool and `Random.nextInt(100) = 95`? In your case the result would be `95, 96, ... 105`, so we get `5` transactions as a result instead of `10`

Reply:
@oskin1 (idx to (idx + qty)).map( _ % max) ?